conwaylib/BoardExtensions.cs

```diff
@@ -3,74 +3,125 @@
-            ++neighbours;
-          }
-        }
-      }
+            //Fast way
+            if (board is IAgeArrayBoard ageArrayBoard)
+            {
+                int?[][] ageArray = ageArrayBoard.GetAgeBoard();
+                if (minY < y && maxY > y && minX < x && maxX > x)
+                {
+                    int?[] minYArray = ageArray[minY];
+                    int?[] yArray = ageArray[y];
+                    int?[] maxYArray = ageArray[maxY];
 
-      return neighbours;
-    }
+                    if (minYArray[minX].HasValue)
+                    {
+                        ++neighbours;
+                    }
+                    if (yArray[minX].HasValue)
+                    {
+                        ++neighbours;
+                    }
+                    if (maxYArray[minX].HasValue)
+                    {
+                        ++neighbours;
+                    }
 
-    /// <summary>
-    /// Randomises the state of all cells in <paramref name="board"/>. The proportion of cells that
-    /// are dead is given by <paramref name="deadFraction"/>, which must be between 0 and 1.
-    /// </summary>
-    public static IMutableBoard Randomise(this IMutableBoard board, Random random, double deadFraction)
-    {
-      for (int x = 0; x < board.Width; ++x)
-      {
-        for (int y = 0; y < board.Height; ++y)
+                    if (minYArray[x].HasValue)
+                    {
+                        ++neighbours;
+                    }
+                    if (maxYArray[x].HasValue)
+                    {
+                        ++neighbours;
+                    }
+
+                    if (minYArray[maxX].HasValue)
+                    {
+                        ++neighbours;
+                    }
+                    if (yArray[maxX].HasValue)
+                    {
+                        ++neighbours;
+                    }
+                    if (maxYArray[maxX].HasValue)
+                    {
+                        ++neighbours;
+                    }
+
+                    return neighbours;
+                }
+            }
+            
+            //Slow way
+            for (int xx = minX; xx <= maxX; ++xx)
+            {
+                for (int yy = minY; yy <= maxY; ++yy)
+                {
+                    if ((xx == x) && (yy == y))
+                    {
+                        continue;
+                    }
+
+                    if (board.Cell(xx, yy))
+                    {
+                        ++neighbours;
+                    }
+                }
+            }
+
+            return neighbours;
+        }
+
+        /// <summary>
+        /// Randomises the state of all cells in <paramref name="board"/>. The proportion of cells that
+        /// are alive is given by <paramref name="liveFraction"/>, which must be between 0 and 1.
+        /// </summary>
+        public static IMutableBoard Randomise(this IMutableBoard board, Random random, double liveFraction)
```

Comment:
Generally I would prefer to see something like this fixed in a separate PR, but not going to insist on it this time.

In Igloo, we have concept of a "Small Fix" whereby a developer can fix some silly little thing without doing the admin of creating an Issue. For such things in the future in our project, you can just make a branch (without issue number in its name) and a PR. Don't fix bugs this way, though - they should have issues to track the fact that we spotted and fixed the bug